### DISCUSSION TITLE
Cloudlog OQRS not working, and QSO save issues

### DISCUSSION THREAD

**Discussion by DeveloperNeon** [magicbug/Cloudlog#2149] [Q&A] [answered]
I recently deployed cloudlog on my webserver, and Ive been having a couple issues.

My first issue occurs with the OQRS system. When visiting https://cloudlogurl.com/index.php/oqrs, everything seems to work as normal. Once you select your station and click proceed though, it brings you to a blank page. A gif has been attached below.

![OQRS Issue](https://github.com/magicbug/Cloudlog/assets/51766327/4636b5e7-33a6-4903-8c03-5d4f81e2d565)

My second issue is with saving QSOs. Ive tried to do this with changing the mode, band, etc, and with using the default options that get set when the page loads. Whenever I try to save a QSO after entering the person's details (usually just the callsign and RST info) and try to click save QSO, it doesnt save. A gif has been attached below.

![QSO Not Saving](https://github.com/magicbug/Cloudlog/assets/51766327/fa85c48a-7c97-459f-b629-c4e3a7d92d7e)

Am I doing anything wrong? Any input would be greatly appreaciated, as I would really like to use Cloudlog to its full potential.

Thanks, 
KD9NSC

**Comment by R1BLH** [comment]
Is any useful information in the logs of the web server and database server?

**Reply by DeveloperNeon** [reply]
I'll check when I get back home

**Reply by DeveloperNeon** [reply]
Sorry for the late reply, there dont seem to be any useful logs.-

**Comment by AndreasK79** [comment]
Hi, update your country files under admin.

**Reply by DeveloperNeon** [reply]
I have tried updating everything, and I'm still having issues.

**Reply by phl0** [reply]
Does it look like this?

![Screenshot from 2023-05-21 20-42-21](https://github.com/magicbug/Cloudlog/assets/7112907/3ff5fbde-8702-4345-96b9-58a2eb762a8d)

(without the xml file error)

**Reply by DeveloperNeon** [reply]
> Does it look like this?
> 
> ![Screenshot from 2023-05-21 20-42-21](https://user-images.githubusercontent.com/7112907/239761610-3ff5fbde-8702-4345-96b9-58a2eb762a8d.png)
> 
> (without the xml file error)

Sorry for the late reply.
It says its updating, then no text appears, then it goes back to updating. This process repeated, and then stopped after a few minutes, and said it was done. A screenshot is attached below.

![image](https://github.com/magicbug/Cloudlog/assets/51766327/aa2824ba-afaf-48a7-b888-be49a0120c33)

**Reply by phl0** [reply]
Ok, thats looks as if DXCC tables are updated. Can you please check the DXCC dropdown after you entered a call and de-focuessed the callsign field (i.e. press TAB). Does it stay empty?

**Reply by DeveloperNeon** [reply]
> Ok, thats looks as if DXCC tables are updated. Can you please check the DXCC dropdown after you entered a call and de-focuessed the callsign field (i.e. press TAB). Does it stay empty?

It does seem to fill in the DXCC dropdown now. The OQRS system still doesnt seem to be working tho.

**Reply by phl0** [reply]
OQRS and QSO logging will not work if DXCC tables are not filled (properly). So the issue is with the update of DXCC data most probably...

**Reply by DeveloperNeon** [reply]
> OQRS and QSO logging will not work if DXCC tables are not filled (properly). So the issue is with the update of DXCC data most probably...

Is there any way I can confirm the DXCC tables are filled properly?

**Reply by phl0** [reply]
> Is there any way I can confirm the DXCC tables are filled properly?

Do you have console access to webroot and SQL db?

**Reply by DeveloperNeon** [reply]
> > Is there any way I can confirm the DXCC tables are filled properly?
> 
> Do you have console access to webroot and SQL db?

I do

**Reply by DeveloperNeon** [reply]
https://mylog.tld/log/index.php/oqrs/get_station_info seems to be returning a 500 error, but my webserver logs dont show anything.

**Reply by DeveloperNeon** [reply]
I have finally gotten a log. It seems to be an issue with the SQL query. 

```
ERROR - 2023-06-11 23:18:02 --> Query error: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ') AND COL_PROP_MODE != "SAT"' at line 1 - Invalid query: SELECT distinct LOWER(`COL_BAND`) as `COL_BAND` FROM `TABLE_HRD_CONTACTS_V01` WHERE station_id in () AND COL_PROP_MODE != "SAT"
ERROR - 2023-06-11 23:18:02 --> Severity: error --> Exception: Call to a member function result() on bool /var/www/web/log/application/models/Bands.php 327
```

It seems like the issue is that no station id is getting passed through.

**Reply by DeveloperNeon** [reply]
I fixed the initial issue (if iI remember to, Ill make a PR), but now Im stuck at the "enter your callsign" page. The function executed when the search button is pressed just returns undefined. When I try sending a post request to the page with the nessisary details, I get a 500 error. If anyone has any ideas, feel free to send them my way.

**Reply by DeveloperNeon** [reply, marked as answer]
I have managed to fix all the issues, and have created a pull request with everything that I did. You can find that pull request [here.](https://github.com/magicbug/Cloudlog/pull/2193) (https://github.com/magicbug/Cloudlog/pull/2193)

**Reply by AndreasK79** [reply]
@DeveloperNeon thanks. It does seem like something has broken OQRS. It was not very long time ago I did test it (and it worked). I tried your PR, but the result were the same.

I will take a look and see if I can figure out where everything went wrong.

**Reply by AndreasK79** [reply]
@DeveloperNeon just to add to it here. Your fix partially works. I have a fix that I think works now. It is based on your PR. I just need @phl0 to verify.
